experiments/pod-delete/pod-delete.sh

```diff
@@ -0,0 +1,126 @@
+#!/bin/sh
+
+set -e
+
+## mandatory arg
+container_id=${CONTAINER_ID}
+interval=${INTERVAL:=5}
+force=${FORCE:=false}
+app_ns=${APP_NS}
+kill_count=${KILL_COUNT}
+app_label=${APP_LABEL}
+duration=${DURATION}
+c_engine=${CHAOS_ENGINE}
+c_ns=${CHAOS_NAMESPACE}
+c_uid=${CHAOS_UID}
+
+## Capture Current time
+startTimeStamp=$(date +%s)
+diffTimeStamp=0
+while [[ "${diffTimeStamp}" -lt "${duration}" ]]
+do
+    currentTimeStamp=$(date +%s)
+    diffTimeStamp=$(( $currentTimeStamp - $startTimeStamp ))
+#############################################################
+###############    CHECKING KILL COUNT     ##################
+#############################################################
+
+## When the kill count is not defined choose any single random pod with the given label and namesapce
+if [[ -z ${kill_count} ]]; then
+    echo "[Inject]: Kill a random application"
+    rand_pod=$(kubectl get pod -n ${app_ns} -l ${app_label} -o=custom-columns=NAME:".metadata.name" --no-headers | shuf -n1)
+    app_pod=${rand_pod}
+else
+    # When kill count is defined select the equal number of pod for chaos with given namespace and label
+    echo "[Inject]: Starting experiment with kill count value: ${kill_count}"
+    pod_list=$(kubectl get pod -n ${app_ns} -l ${app_label} -o=custom-columns=NAME:".metadata.name" --no-headers | shuf -n${kill_count})
+    app_pod=${pod_list}
+fi
+
+#############################################################
+###############    GENERATING EVENTS     ####################
+#############################################################
+if [[ ! -z ${c_engine} ]]; then
+    NOW=$( date -u +"%Y-%m-%dT%H:%M:%SZ" )
+    jinja2 -D engine_ns=${c_ns} -D ts=${NOW} -D engine_name=${c_engine} -D engine_uid=${c_uid} pod-delete-event.yaml > helper-pod.yaml
```

Comment:
the source shouldn't be ""helper-pod" -- it should be the name of this pod (launched by the experiment) !! So we need to get it via a downward API. cc: @ispeakc0de

Reply:
Addressed:
```bash
Events:
  Type    Reason                     Age   From                     Message
  ----    ------                     ----  ----                     -------
  Normal  ChaosEngineInitialized     56s   chaos-operator           nginx-chaos-runner created successfully
  Normal  ExperimentDependencyCheck  53s   nginx-chaos-runner       Experiment resources validated for Chaos Experiment: 'pod-delete'
  Normal  ExperimentJobCreate        48s   nginx-chaos-runner       Experiment Job 'pod-delete-a447z9' created for Chaos Experiment 'pod-delete'
  Normal  PreChaosCheck              32s   pod-delete-a447z9-vdl2w  AUT is Running successfully
  Normal  ChaosInject                3s    pod-delete-ixhaqh        Killing the application pod "nginx-7db9fccd9b-h4qmk"
```